### ISSUE TITLE
xgb for multiclass requires classes to be integers enumerated from 0

### ISSUE THREAD

**Issue by panas89** [uclamii/model_tuner#107]
- xgboost library expects class to be integer from 0
- sklearn and catboost does not expect this, works with all types of class
- Maybe explain in documentation that class variable has to be encoded or provide this functionality

**Comment by lshpaner** [comment]
Overarching report metrics output issue has been extensively mitigated through https://github.com/uclamii/model_tuner/pull/128. Moreover, no such issues have been detected when testing `xgb_multi.py`. When re-encoding class labels to numbers `class_labels=["1", "2", "3"]`, and the model is run, the following output is displayed:

```bash
Model Tuner version: 0.0.23a
Model Tuner authors: Arthur Funnell, Leonid Shpaner, Panayiotis Petousis


Pipeline Steps:

┌───────────────────────────────────────────────────────────┐
│ Step 1: preprocess_column_transformer_ColumnTransformer   │
│ ColumnTransformer                                         │
└───────────────────────────────────────────────────────────┘
                             │
                             ▼
┌───────────────────────────────────────────────────────────┐
│ Step 2: xgb_mc                                            │
│ XGBClassifier                                             │
└───────────────────────────────────────────────────────────┘

100%|████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████| 12/12 [01:17<00:00,  6.42s/it]
Best score/param set found on validation set:
{'params': {'xgb_mc__early_stopping_rounds': 20,
            'xgb_mc__eval_metric': 'mlogloss',
            'xgb_mc__max_depth': 10,
            'xgb_mc__n_estimators': 10},
 'score': 0.9666666666666668}
Best roc_auc_ovr: 0.967 

Validation Metrics
--------------------------------------------------------------------------------
1
          Predicted:
            Pos  Neg
--------------------------------------------------------------------------------
Actual: Pos 19 (tn)   1 (fp)
        Neg  0 (fn)  10 (tp)
--------------------------------------------------------------------------------
2
          Predicted:
            Pos  Neg
--------------------------------------------------------------------------------
Actual: Pos 19 (tn)   1 (fp)
        Neg  2 (fn)   8 (tp)
--------------------------------------------------------------------------------
3
          Predicted:
            Pos  Neg
--------------------------------------------------------------------------------
Actual: Pos 19 (tn)   1 (fp)
        Neg  1 (fn)   9 (tp)
--------------------------------------------------------------------------------

              precision    recall  f1-score   support

           1       0.91      1.00      0.95        10
           2       0.89      0.80      0.84        10
           3       0.90      0.90      0.90        10

    accuracy                           0.90        30
   macro avg       0.90      0.90      0.90        30
weighted avg       0.90      0.90      0.90        30

--------------------------------------------------------------------------------
Test Metrics
--------------------------------------------------------------------------------
1
          Predicted:
            Pos  Neg
--------------------------------------------------------------------------------
Actual: Pos 18 (tn)   2 (fp)
        Neg  0 (fn)  10 (tp)
--------------------------------------------------------------------------------
2
          Predicted:
            Pos  Neg
--------------------------------------------------------------------------------
Actual: Pos 19 (tn)   1 (fp)
        Neg  2 (fn)   8 (tp)
--------------------------------------------------------------------------------
3
          Predicted:
            Pos  Neg
--------------------------------------------------------------------------------
Actual: Pos 20 (tn)   0 (fp)
        Neg  1 (fn)   9 (tp)
--------------------------------------------------------------------------------

              precision    recall  f1-score   support

           1       0.83      1.00      0.91        10
           2       0.89      0.80      0.84        10
           3       1.00      0.90      0.95        10

    accuracy                           0.90        30
   macro avg       0.91      0.90      0.90        30
weighted avg       0.91      0.90      0.90        30

--------------------------------------------------------------------------------
********************************************************************************
Report Model Metrics: xgb_mc

           Class     Metric     Value
0              1  Precision  0.833333
1              1     Recall  1.000000
2              1   F1-Score  0.909091
3              2  Precision  0.888889
4              2     Recall  0.800000
5              2   F1-Score  0.842105
6              3  Precision  1.000000
7              3     Recall  0.900000
8              3   F1-Score  0.947368
9   Weighted Avg  Precision  0.907407
10  Weighted Avg     Recall  0.900000
11  Weighted Avg   F1-Score  0.899522
12  Weighted Avg   Accuracy  0.900000
********************************************************************************
```
